File: incidentAssistant/app/channels/suggestions_channel.rb
class SuggestionsChannel < ApplicationCable::Channel
  def subscribed
    incident_id = params[:incident_id]
    stream_from "incident_#{incident_id}_suggestions"
    
    Rails.logger.info "Client subscribed to incident #{incident_id} suggestions"
  end
  
  def unsubscribed
    Rails.logger.info "Client unsubscribed from suggestions channel"
  end
end

File: incidentAssistant/app/javascript/channels/suggestions_channel.js
import consumer from "./consumer";

export function subscribeTo(incidentId, callbacks) {
  return consumer.subscriptions.create(
    { channel: "SuggestionsChannel", incident_id: incidentId },
    {
      connected() {
        console.log('Connected to SuggestionsChannel for incident', incidentId);
      },
      
      disconnected() {
        console.log('Disconnected from SuggestionsChannel');
      },
      
      received(data) {
        console.log('Received data:', data);
        if (callbacks.received) {
          callbacks.received(data);
        }
      }
    }
  );
}
